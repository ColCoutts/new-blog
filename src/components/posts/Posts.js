import React from 'react';
import PropTypes from 'prop-types';
import Post from './Post';

function Posts({ posts, deletePost }) {
  const postList = posts.map((post, id) => (
    <li key={id}>
      <Post post={post} deletePost={deletePost} />
    </li>
  ));
  
  return (
    <ul>
      {postList}
    </ul>
  );
}

Posts.propTypes = {
  deletePost: PropTypes.func.isRequired,
  posts: PropTypes.array.isRequired
};

export default Posts;
