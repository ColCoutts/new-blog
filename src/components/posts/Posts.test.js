import React from 'react';
import { shallow } from 'enzyme';
import Posts from './Posts';

describe('posts snapshot test', () => {
  it('renders Posts snapshot', () => {
    const newArray = [
      {
        title: 'hi',
        body: 'test'
      },
      {
        title: 'hi2',
        body: 'test2'
      }
    ];
    const wrapper = shallow(<Posts posts={newArray} deletePost={ () => {} } />);
    expect(wrapper).toMatchSnapshot();
  });
});
