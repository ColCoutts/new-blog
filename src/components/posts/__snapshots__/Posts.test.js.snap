// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`posts snapshot test renders Posts snapshot 1`] = `
<ul>
  <li
    key="0"
  >
    <Post
      deletePost={[Function]}
      post={
        Object {
          "body": "test",
          "title": "hi",
        }
      }
    />
  </li>
  <li
    key="1"
  >
    <Post
      deletePost={[Function]}
      post={
        Object {
          "body": "test2",
          "title": "hi2",
        }
      }
    />
  </li>
</ul>
`;
